File: example/java/spring-boot/two_consumers_one_provider/legal-person-consumer/src/main/java/br/com/zup/pact/legalperson/integration/account/service/impl/AccountIntegrationServiceImpl.java
package br.com.zup.pact.legalperson.integration.account.service.impl;

import br.com.zup.pact.legalperson.dto.BalanceDTO;
import br.com.zup.pact.legalperson.integration.account.service.AccountIntegrationService;
import java.util.Optional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

@Service
public class AccountIntegrationServiceImpl implements AccountIntegrationService {

    @Autowired
    private RestTemplate restTemplate;

    @Value("${integration.account.balance.url}")
    private String accountBalanceUrl;

    @Override
    public Optional<BalanceDTO> getBalance(Integer accountId) {
        final ResponseEntity<BalanceDTO> responseEntity = restTemplate.getForEntity(accountBalanceUrl, BalanceDTO.class, accountId);
        return Optional.ofNullable(responseEntity.getBody());
    }
}

File: example/kotlin/one-consumer-one-provider/consumer/consumer/src/main/kotlin/br/com/zup/pact/consumer/integration/account/service/impl/AccountIntegrationServiceImpl.kt
package br.com.zup.pact.consumer.integration.account.service.impl

import br.com.zup.pact.consumer.dto.BalanceDTO
import br.com.zup.pact.consumer.integration.account.service.AccountIntegrationService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.http.ResponseEntity
import org.springframework.web.client.RestTemplate

class AccountIntegrationServiceImpl(
        @Autowired
        private val restTemplate: RestTemplate,

        @Value("integration.account.balance.url")
        private val accountBalanceUrl: String

): AccountIntegrationService {

    override fun getBalance(accountId: Int): BalanceDTO? {
        val responseEntity: ResponseEntity<BalanceDTO> = restTemplate.getForEntity(accountBalanceUrl,
        BalanceDTO::class.java, accountId)
        return responseEntity.body
    }
}